distributed/worker.py

```diff
@@ -2338,8 +2317,8 @@ def rescind_key(self, key):
 
     # FIXME: this breaks if changed to async def...
     # xref: https://github.com/dask/distributed/issues/3938
-    @gen.coroutine
-    def executor_submit(self, key, function, args=(), kwargs=None, executor=None):
+    # @gen.coroutine
+    async def executor_submit(self, key, function, args=(), kwargs=None, executor=None):
```

Comment:
This should not be part of this PR. I put this in while debugging the deadlock, trying to figure out if it was a problem with the threadpool. Changes might be viable anyhow but should be part of another PR

- [ ] PR for executor_submit changes